unistd/pwd.c

```diff
@@ -24,69 +24,94 @@
 
 struct passwd pwnam;
 char pw_name[NAME_MAX];
+char intstr[16];
 char pw_gecos[128];
 char pw_dir[PATH_MAX];
 char pw_shell[PATH_MAX];
 char pw_passwd[128];
 
 
-struct passwd *getpwnam(const char *name)
+static void readpwentry(FILE *fp, char *pwentry, unsigned int maxlgth)
 {
-	FILE *fp;
-	char buf[1024];
+	char c = 0;
+	unsigned int i = 0;
+	c = fgetc(fp);
+	while (c != ':' && c != '\n' && c != EOF) {
+		if (i < maxlgth)
```

Comment:
```suggestion
		if (i < maxlgth - 1)
```
You need space for the `null` character.
The question is what if `maxlgth == 0`? Function is local and perhaps such a case can be ruled out.

Reply:
- Thanks, I missed that one on `maxlgth - 1`. Added `-1`
 - this function exists only for not writing _read operation_ for each `passwd` field so it is intentionally designed to not be versatile across whole input variables space.  If there is no use case for it outside I would rather left it as is.